### PR TITLE
Add testing for clinical ingest

### PR DESCRIPTION
**Pull request by justin-ys**
- This PR adds implementation testing for each Clinical Ingest field using pytest, ensuring they are all added correctly by Katsu.
- Tests can be run by invoking test.py in the root directory.

NOTE: A lot of code is refactored, but it still works well, I think. The pytest test only checks internal stuff, to make sure that single_ingest.json is flattened correctly for ingest into Katsu. The other tests will be rolled into integration testing in the CanDIGv2 repo.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ARG venv_python
 FROM python:${venv_python}
 
+ARG prod_environment=FALSE
+ENV PROD_ENVIRONMENT=${prod_environment}
+
+ARG prod_environment=FALSE
+ENV PROD_ENVIRONMENT=${prod_environment}
```
Comment (SonQBChau) [inline Dockerfile]: is this a duplicate?

File: katsu_ingest.py [hunk 2]
```diff
@@ -64,42 +66,50 @@ def read_json(file_path):
...
+            update_headers(headers)
+            response = requests.post(
+                ingest_url, headers=headers, data=json.dumps(fields[type])
+            )
+
+            if response.status_code == HTTPStatus.CREATED:
+                result["results"].append(f"Of {len(fields[type])} {type}, {response.json()['result']} were created")
+            elif response.status_code == HTTPStatus.NOT_FOUND:
+                message = f"ERROR 404: {ingest_url} was not found! Please check the URL."
+                result["errors"].append(f"{type}: message")
```
Comment (SonQBChau) [inline katsu_ingest.py]: is this a typo? Should we log the message also?

File: katsu_ingest.py [hunk 3]
```diff
@@ -64,42 +66,50 @@ def read_json(file_path):
...
+                break
+            elif response.status_code == HTTPStatus.FORBIDDEN:
+                message = f"ERROR 403: You do not have permission to ingest {type} for {fields[type][0]['program_id']}"
+                result["errors"].append(f"{type}: message")
+                break
+            else:
+                message = f"\nREQUEST STATUS CODE: {response.status_code} \nRETURN MESSAGE: {response.text}\n"
```
Comment (SonQBChau) [inline katsu_ingest.py]: same as above, if the message is unused, can we delete it?

**Comment by SonQBChau** [comment]
test_flatten.py should be removed or updated to pass the tests

**Review by SonQBChau — APPROVED** [review]
some minor things, otherwise looks good

File: Dockerfile (hunk 1 above)
Comment (daisieh) [reply Dockerfile]: looks like it...no idea what this is about.

File: katsu_ingest.py (hunk 2 above)
Comment (daisieh) [reply katsu_ingest.py]: sure, fixing this

File: katsu_ingest.py (hunk 3 above)
Comment (daisieh) [reply katsu_ingest.py]: fixing this

**Pull request merged**